salt/utils/event.py

```diff
@@ -1075,24 +1075,39 @@ def _handle_signals(self, signum, sigframe):
         super(EventReturn, self)._handle_signals(signum, sigframe)
 
     def flush_events(self):
-        event_return = '{0}.event_return'.format(
-            self.opts['event_return']
-        )
+        if isinstance(self.opts['event_return'], list):
+            # Multiple event returners
+            for r in self.opts['event_return']:
+                if log.level <= logging.DEBUG:
+                    log.debug('Calling event returner {0}, one of many.'.format(r))
+                event_return = '{0}.event_return'.format(r)
+                self._flush_event_single(event_return)
+        else:
+            # Only a single event returner
+            if log.level <= logging.DEBUG:
```

Comment:
CNP from a few lines down, but I see that there was a particular reason there. Removed.